src/languages/lsl.js

```diff
@@ -0,0 +1,133 @@
+/*
+    Language: Linden Scripting Language
+    Description: The Linden Scripting Language is used in Second Life by Linden Labs.
+    Author: Builder's Brewery <buildersbrewery@gmail.com>
+    Category: common, scripting
+*/
+
+function(hljs) {
+    return {
+        case_insensitive: false,
+        aliases: ['lsl'],
+        illegal: ':',
+        contains: [
+            {
+                className: 'string',
+                begin: '"',
+                end: '"'
+            },
+            {
+                className: 'comment',
+                begin: '//',
+                end: '$'
+            },
+            {
+                className: 'comment',
+                begin: '/\\*',
+                end: '\\*/'
+            },
+            {
+                className: 'number',
+                begin: '(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)'
+            },
```

Comment:
Looks like one of the standard modes may work here as well: https://github.com/isagalaev/highlight.js/blob/master/src/highlight.js#L726-L740